File: src/org/usfirst/frc/team6325/robot/commands/Auto/TurnToAngle.java

package org.usfirst.frc.team6325.robot.commands.Auto;

import org.usfirst.frc.team6325.robot.Robot;

import edu.wpi.first.wpilibj.command.Command;

public class TurnToAngle extends Command {

	double ang = 0.0;
	boolean done = false;
	
	public TurnToAngle(double angle) {
		requires(Robot.drivetrain);
		ang = angle;
		
	}
	
	protected void initialize() {
    	Robot.drivetrain.autoTurnInit(ang);
    }

    // Called repeatedly when this Command is scheduled to run
    protected void execute() {
    	Robot.drivetrain.autoTurn();
    	
    	if(Math.abs(Robot.drivetrain.getAngle() - ang) <= 5) {
    		done = true;
    	}
    }

    // Make this return true when this Command no longer needs to run execute()
    protected boolean isFinished() {
        return done;
    }

    // Called once after isFinished returns true
    protected void end() {
    	Robot.drivetrain.autoTurnStop();
    	Robot.drivetrain.killMotors();
    }

    // Called when another command which requires one or more of the same
    // subsystems is scheduled to run
    protected void interrupted() {
    	end();
    }

}

File: src/main/java/org/usfirst/frc/team6325/robot/commands/Auto/TurnToAngle.kt
package org.usfirst.frc.team6325.robot.commands.Auto

import org.usfirst.frc.team6325.robot.Polybius
import org.sertain.command.Command


public class TurnToAngle : Command
{
    var ang: Double = 0.0; var diff: Double = 0.0; var rate: = Double = 0.3
    var done: Boolean = false

    public fun TurnToAngle(angle: Double)
    {
        requires(Polybius.drivetrain)
        ang = angle
    }

    init()
    {
        diff = ang - Polybius.drivetrain.getAngle()
        
        if(diff > 0)
            Polybius.drivetrain.drive(rate, -rate)
        else
            Polybius.drivetrain.drive(-rate, rate)
    }

    override fun execute()
    {
        if (diff > 0)
        {
            if (Polybius.drivetrain.getAngle() > ang)
                done = true
        }
        else
        {
            if (ang > Polybius.drivetrain.getAngle())
                done = true
        }
    }
    
    override fun isCompleted(): Boolean = return done
    override fun isCanceled() = onDestroy()

    override fun onDestroy()
    {
        Polybius.drivetrain.autoTurnStop()
        Polybius.drivetrain.killMotors()
    }
}